nomad/structs/services.go

```diff
@@ -1545,6 +1553,8 @@ func (t *SidecarTask) Copy() *SidecarTask {
 
 	nt.VolumeMounts = CopySliceVolumeMount(t.VolumeMounts)
 
+	nt.Identities = CopySliceWorkloadIdentity(t.Identities)
```

Comment:
We have a generic implementation we can use for anything that implements the `helper.Copy` interface, which WIs do

```suggestion
	nt.Identities = helper.CopySlice(t.Identities)
```